### ISSUE TITLE
`toDate` returns unexpected date in America/Sao_Paolo timezone.

### ISSUE THREAD

**Issue by inytar** [date-fns/date-fns#992]
On daylight switching (2018-11-04) in Sao Paolo Brazil the `toDate` function returns a different date object then expected:

```javascript
const toDate = require('date-fns/toDate');
const getDay = require('date-fns/getDay');

const nativeSunday = new Date(2018, 10, 4)
console.log(nativeSunday)
// 2018-11-04T03:00:00.000Z
console.log(getDay(nativeSunday))
// 0
const dateFnsSunday = toDate('2018-11-04')
console.log(dateFnsSunday)
// 2018-11-04T02:00:00.000Z
console.log(getDay(dateFnsSunday))
// 6
```
This problem only manifests itself if running in the Sao Paolo timezone. This can be tested by running:
```sh
env TZ='America/Sao_Paolo' node testScript.js
```

**Comment by inytar** [comment]
#972 shows a similar issue for the formatting function.

**Comment by kossnocorp** [comment]
This fix for this problem was released with `v2.0.0-alpha.26` and `v1.30.1`. 

See: 

- https://blog.date-fns.org/post/v200-alpha26-is-out-2rizgx9wk2g1t
- https://blog.date-fns.org/post/v130-is-out-m35nqh4m13r7
